app/controllers/api/service_orders_controller.rb

```diff
@@ -60,6 +60,15 @@ def add_request_to_cart(workflow)
       workflow.add_request_to_cart
     end
 
+    def create_service_order(data)
+      ServiceOrder.new(data).tap do |service_order|
+        service_order.assign_user
+        service_order.save!
+      end
```

Comment:
can't the above be replaced with ServiceOrder.create(data) ?  we might get assign_user with that as it seems to be a before create action.

Reply:
@abellotti that's what I was hoping for, but unfortunately the assign_user happens after the validation. I played around with the model a bit but was unable to update it without breaking anything else. I can play around with it more if you would prefer that route.

Reply:
Ahh ok, no worries. this is ok for now.